contracts/Staking.sol

```diff
@@ -113,148 +96,181 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
+        // No need to check that have enough locked funds, as _decreaseActiveLockAmount will fail
+
+        _transfer(_from, msg.sender, _to, _toLockManager, _amount);
+        _decreaseActiveLockAmount(_from, msg.sender, _amount);
+    }
+
+    /**
+     * @notice Allow `_lockManager` to lock up to `_allowance` allowed tokens of `msg.sender`
+     *         It creates a new inactive lock, so the lock for this manager cannot exist before.
+     * @param _lockManager The manager entity for this particular lock
+     * @param _allowance Amount of allowed tokens increase
+     */
+    function allowNewLockManager(address _lockManager, uint256 _allowance) external isInitialized {
+        Lock storage lock_ = accounts[msg.sender].locks[_lockManager];
+        require(lock_.allowance == 0, ERROR_LOCK_ALREADY_EXISTS);
+
+        _increaseLockAllowance(_lockManager, lock_, _allowance);
+    }
 
-        _transfer(_from, _fromLockId, _to, _toLockId, _amount);
-        _updateActiveLockAmount(_from, _fromLockId, _amount, false);
+    /**
+     * @notice Increase `_allowance` allowed tokens for lock of `msg.sender` by `_lockManager`
+     * @param _lockManager The manager entity for this particular lock
+     * @param _allowance Amount of allowed tokens increase
+     */
+    function increaseLockAllowance(address _lockManager, uint256 _allowance) external isInitialized {
+        Lock storage lock_ = accounts[msg.sender].locks[_lockManager];
+        require(lock_.allowance > 0, ERROR_LOCK_DOES_NOT_EXIST);
+
+        _increaseLockAllowance(_lockManager, lock_, _allowance);
     }
 
     /**
-     * @notice Decrease the amount of tokens locked in `_accountAddress`'s lock #`_lockId` to `_newAmount`
+     * @notice Decrease `_allowance` allowed tokens for lock of `_accountAddress` by `_lockManager`
      * @param _accountAddress Owner of locked tokens
-     * @param _lockId Id of the lock for the given account
-     * @param _newAmount New amount of locked tokens
+     * @param _lockManager The manager entity for this particular lock
+     * @param _allowance Amount of allowed tokens decrease
      */
-    function decreaseLockAmount(address _accountAddress, uint256 _lockId, uint256 _newAmount) external isLockManager(_accountAddress, _lockId) {
-        // lock 0 tokens makes no sense
-        require(_newAmount > 0, ERROR_AMOUNT_ZERO);
+    function decreaseLockAllowance(address _accountAddress, address _lockManager, uint256 _allowance) external isInitialized {
+        // only owner and manager can decrease allowance
+        require(msg.sender == _accountAddress || msg.sender == _lockManager, ERROR_CANNOT_CHANGE_ALLOWANCE);
+        require(_allowance > 0, ERROR_AMOUNT_ZERO);
+
+        Lock storage lock_ = accounts[_accountAddress].locks[_lockManager];
+        uint256 newAllowance = lock_.allowance.sub(_allowance);
+        require(newAllowance >= lock_.amount, ERROR_NOT_ENOUGH_ALLOWANCE);
```

Comment:
Would it be harmful if we support reducing the allowance after your tokens were locked by the manager?

Reply:
I don’t like the idea, it feels like an inconsistent state to me.

Reply:
Actually, see my related response here: https://github.com/aragon/staking/pull/16#discussion_r428858624